iothub/device/src/Exceptions/IotHubErrorCode.cs

```diff
@@ -6,7 +6,7 @@ namespace Microsoft.Azure.Devices.Client.Exceptions
     /// <summary>
     /// The IoT Hub status code.
     /// </summary>
-    public enum IotHubStatusCode
+    public enum IotHubErrorCode
```

Comment:
I'd like us to avoid naming conflicts with the service SDK. Can we make these distinct somehow?